packages/core/src/tracing/trace.ts

```diff
@@ -215,6 +217,47 @@ export const continueTrace = <V>(
   }
 
   const { sentryTrace, baggage } = options;
+  const client = getClient();
+
+  const clientOptions = client?.getOptions();
+  const strictTraceContinuation = clientOptions?.strictTraceContinuation || false; // default for `strictTraceContinuation` is `false` todo(v10): set default to `true`
+
+  const incomingDsc = baggageHeaderToDynamicSamplingContext(baggage);
+  const baggageOrgId = incomingDsc?.org_id;
+
+  let sdkOrgId: string | undefined;
+  const dsn = client?.getDsn();
+  if (dsn?.host) {
+    sdkOrgId = extractOrgIdFromDsnHost(dsn.host);
+  }
+
+  const shouldStartNewTrace = (): boolean => {
```

Comment:
l: Let's maybe make this a separate function that takes arguments instead of inlining this function? No strong feelings, but this is how we usually do this I suppose 🤔

Reply:
yeah no strong opinions about this. Just thought it's not a lot of logic and when inlining this it does not disrupt you from the code-reading flow but it still keeps this part of the code separated :D